src/Adaptivity/AdaptedDiscreteModels.jl

```diff
@@ -0,0 +1,281 @@
+
+"""
+  AdaptedDiscreteModel
+
+  `DiscreteModel` created by refining/coarsening another `DiscreteModel`. 
+  The refinement/coarsening hierarchy can be traced backwards by following the 
+  `parent` pointer chain. This allows the transfer of dofs 
+  between `FESpaces` defined on this model and its ancestors.
+
+"""
+struct AdaptedDiscreteModel{Dc,Dp,A<:DiscreteModel{Dc,Dp},B<:DiscreteModel{Dc,Dp},C<:AdaptivityGlue} <: DiscreteModel{Dc,Dp}
+  model  ::A
+  parent ::B
+  glue   ::C
+
+  function AdaptedDiscreteModel(model::DiscreteModel{Dc,Dp},parent,glue) where {Dc,Dp}
+    @check !isa(model,AdaptedDiscreteModel)
+    A = typeof(model)
+    B = typeof(parent)
+    C = typeof(glue)
+    return new{Dc,Dp,A,B,C}(model,parent,glue)
+  end
+end
+
+# DiscreteModel API
+Geometry.get_grid(model::AdaptedDiscreteModel)          = get_grid(model.model)
+Geometry.get_grid_topology(model::AdaptedDiscreteModel) = get_grid_topology(model.model)
+Geometry.get_face_labeling(model::AdaptedDiscreteModel) = get_face_labeling(model.model)
+
+# Other getters
+get_model(model::AdaptedDiscreteModel)  = model.model
+get_parent(model::AdaptedDiscreteModel{Dc,Dp,A,<:AdaptedDiscreteModel}) where {Dc,Dp,A} = get_model(model.parent)
+get_parent(model::AdaptedDiscreteModel{Dc,Dp,A,B}) where {Dc,Dp,A,B} = model.parent
+get_adaptivity_glue(model::AdaptedDiscreteModel) = model.glue
+
+# Relationships
+function is_child(m1::AdaptedDiscreteModel,m2::DiscreteModel)
+  return get_parent(m1) === m2 # m1 = refine(m2)
+end
+
+function is_child(m1::AdaptedDiscreteModel,m2::AdaptedDiscreteModel)
+  return get_parent(m1) === get_model(m2) # m1 = refine(m2)
+end
+
+is_child(m1::DiscreteModel,m2::AdaptedDiscreteModel) = false
+
+is_related(m1::AdaptedDiscreteModel,m2::DiscreteModel) = is_child(m1,m2)
+is_related(m1::DiscreteModel,m2::AdaptedDiscreteModel) = is_child(m2,m1)
+is_related(m1::AdaptedDiscreteModel,m2::AdaptedDiscreteModel) = is_child(m1,m2) || is_child(m2,m1)
+
+# Model Refining
+function refine(model::DiscreteModel,args...;kwargs...) :: AdaptedDiscreteModel
```

Comment:
Good call here on the use of the type assertion operator. Well done.

Reply:
Should we actually call this method `adapt` instead of `refine` ?

Reply:
🥇

Reply:
ok, fair enough. Dow we add an `adapt`  `@abstractmethod` ?

Reply:
Done in [e392d98](https://github.com/gridap/Gridap.jl/pull/838/commits/e392d98f248612b9df2cc2e746b8b87e8631f6cf)